src/QsCompiler/CompilationManager/CompilationUnit.cs

```diff
@@ -109,14 +148,22 @@ internal References CombineWith(References other, Action<ErrorCode, string[]> on
 
         /// <summary>
         /// Given a dictionary that maps the ids of dll files to the corresponding headers,
-        /// initializes a new set of references based on the given headers and verifies that there are no conflicts. 
-        /// Calls the given Action onError with suitable diagnostics if two or more references conflict, 
-        /// i.e. if two or more references contain a declaration with the same fully qualified name. 
-        /// Throws an ArgumentNullException if the given dictionary of references is null. 
+        /// initializes a new set of references based on the given headers and verifies that there are no conflicts.
+        /// Calls the given Action onError with suitable diagnostics if two or more references conflict,
+        /// i.e. if two or more references contain a public declaration with the same fully qualified name.
+        /// Throws an ArgumentNullException if the given dictionary of references is null.
         /// </summary>
         public References(ImmutableDictionary<NonNullable<string>, Headers> refs, Action<ErrorCode, string[]> onError = null)
         {
-            this.Declarations = refs ?? throw new ArgumentNullException(nameof(refs));
+            // TODO: We mangle the names of internal declarations in references to avoid name conflicts in the symbol
```

Comment:
Do you think this would be good to do in the future? The name mangling needs to happen at some point, even if the symbol table is more flexible. What would be the reason to do this TODO nonetheless?

Reply:
Is name mangling needed when targeting C# or .NET? Since .NET supports internal classes, I thought the only issue is during Q# symbol resolution.

Reply:
Ah, I was thinking more if we don't want to rely on C#/dotnet.

Reply:
I think would still prefer keeping the name mangling contained to the part that generates the code or IR for the platform being targeted, since the requirements for how names should be mangled depends on the target platform. For example, my understanding (I could be wrong, not super familiar with C++) is that C++ has to mangle the names of namespaces into class/function names, since namespaces do not exist in the C-style library files that C++ compiles into. So depending on what platform we target we may want to mangle names differently or not at all.

Also, if for example we support function overloading (#29), then it might be easier to allow redeclarations of a function anyway, since name mangling early enough in the complication process to avoid name conflicts there seems difficult, potentially.

Reply:
Sure, I agree. For now, I would draw the line at having something to support name mangling in general and a todo here. :)